src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Char.cs

```diff
@@ -71,63 +71,200 @@ public static unsafe int SequenceCompareTo(ref char first, int firstLength, ref
             Debug.Assert(firstLength >= 0);
             Debug.Assert(secondLength >= 0);
 
-            int lengthDelta = firstLength - secondLength;
-
             if (Unsafe.AreSame(ref first, ref second))
                 goto Equal;
 
-            IntPtr minLength = (IntPtr)((firstLength < secondLength) ? firstLength : secondLength);
-            IntPtr i = (IntPtr)0; // Use IntPtr for arithmetic to avoid unnecessary 64->32->64 truncations
+            nint minLength = Math.Min(firstLength, secondLength);
+            nint offset = 0;
 
-            if ((byte*)minLength >= (byte*)(sizeof(UIntPtr) / sizeof(char)))
+            if (Sse2.IsSupported && minLength >= Vector128<ushort>.Count)
             {
-                if (Vector.IsHardwareAccelerated && (byte*)minLength >= (byte*)Vector<ushort>.Count)
+                // >= Sse2 intrinsics are supported, and length is enough to use them so use that path.
+                // We jump forward to the intrinsics at the end of them method so a naive branch predict
+                // will choose the non-intrinsic path so short lengths which don't gain anything aren't
+                // overly disadvantaged by having to jump over a lot of code. Whereas the longer lengths
+                // more than make this back from the intrinsics.
+                goto IntrinsicsCompare;
+            }
+
+            // Only use the Vector path if intrinsics aren't supported.
```

Comment:
Why is this?  If Sse2.IsSupported but minLength < `Vector128<ushort>.Count`, there's no benefit to using the `Vector<ushort>` path?

Reply:
No because `Vector<ushort>` will be the same size or larger. Is mostly a path for Arm now